code/game/gamemodes/wizard/artefact.dm

```diff
@@ -173,3 +173,254 @@
 	H.equip_to_slot_or_del(new /obj/item/weapon/claymore(H), slot_r_hand)
 	H.equip_to_slot_or_del(new /obj/item/weapon/twohanded/spear(H), slot_back)
+
+/obj/item/weapon/multisword
+	name = "multiverse sword"
+	desc = "A weapon capable of summoning copies of yourself from across space and time to fight by your side."
+	icon_state = "claymore"
+	item_state = "claymore"
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	flags = CONDUCT
+	slot_flags = SLOT_BELT
+	force = 20
+	throwforce = 10
+	w_class = 2
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	var/faction = list("unassigned")
+	var/charged = 40
+	var/assigned = "unassigned"
+	var/evil = TRUE
+
+/obj/item/weapon/multisword/New()
+	..()
+	SSobj.processing |= src
+
+
+/obj/item/weapon/multisword/Destroy()
+	SSobj.processing.Remove(src)
+	..()
+
+
+/obj/item/weapon/multisword/process()
+	if(charged < 40)
+		charged++
+	return
+
+/obj/item/weapon/multisword/attack_self(mob/user)
+	if(user.mind.special_role == "apprentice")
+		user << "<span class='warning'>You know better than to touch your teacher's stuff.</span>"
+		return
+	if(charged == 40)
+		var/faction_check = 0
+		for(var/F in faction)
+			if(F in user.faction)
+				faction_check = 1
+				break
+		if(faction_check == 0)
+			faction = list("[user.real_name]")
+			assigned = "user.realname"
+			user.faction = list("[user.real_name]")
+			user << "<span class='warning'>You bind the sword to yourself. You can now use it to summon help.</span>"
+			if(!usr.mind.special_role)
+				if(prob(30))
+					user << "<span class='warning'>With your new found power you could easily conquer the station!</span>"
+					var/datum/objective/hijackclone/hijack_objective = new /datum/objective/hijackclone
+					hijack_objective.owner = usr.mind
+					usr.mind.objectives += hijack_objective
+					hijack_objective.explanation_text = "Ensure only [usr.real_name] and their copies are on the shuttle!"
+					ticker.mode.traitors += usr.mind
+					usr.mind.special_role = "[usr.real_name] Prime"
+					evil = TRUE
+				else
+					user << "<span class='warning'>With your new found power you could easily defend the station!</span>"
+					var/datum/objective/survive/new_objective = new /datum/objective/survive
+					new_objective.owner = usr.mind
+					new_objective.explanation_text = "Survive, and help defend the innocent from the mobs of multiverse clones."
+					usr.mind.objectives += new_objective
+					ticker.mode.traitors += usr.mind
+					usr.mind.special_role = "[usr.real_name] Prime"
+					evil = FALSE
+		else
+			var/list/candidates = get_candidates(BE_WIZARD)
+			if(candidates.len)
+				var/client/C = pick(candidates)
+				spawn_copy(C, get_turf(user.loc), user)
+				user << "<span class='warning'>The sword flashes, and you find yourself face to face with...you!</span>"
+				charged = 0
+				spawn_copy(C, get_turf(user.loc), user)
+				for(var/obj/item/weapon/multisword/M in world)

```

Comment:
I think there's a list for processing objects somewhere in the codebase, or you could define a global list specifically for these swords. Anything's better than a world call.